### PR TITLE
Add tests for IntegerLiteralExprSyntax

### PR DESCRIPTION
**Pull request by harlanhaskins**
This path adds some simple tests to `IntegerLiteralExprSyntax`, fixes a typo in `SyntaxFactory::makePrefixOperator`, and implements
`IntegerLiteralExprSyntax::withSign`.

Resolves #46640

### CODE REVIEW

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Comment by harlanhaskins** [comment]
@bitjammer Mind doing a quick review? Thanks!

File: include/swift/Syntax/SyntaxFactory.h [hunk 1]
```diff
@@ -419,7 +419,7 @@ struct SyntaxFactory {
 #pragma mark - Operators
 
   /// Make a prefix operator with the given text.
-  static RC<TokenSyntax> makePrefixOpereator(OwnedString Name,
```
Comment (bitjammer) [inline include/swift/Syntax/SyntaxFactory.h]: Heh, thanks! Whoops

File: lib/Syntax/ExprSyntax.cpp [hunk 2]
```diff
@@ -114,6 +114,13 @@ IntegerLiteralExprSyntax::withDigits(RC<TokenSyntax> NewDigits) const {
                                                       Cursor::Digits);
 }
 
+IntegerLiteralExprSyntax
+IntegerLiteralExprSyntax::withSign(RC<swift::syntax::TokenSyntax> NewSign)
```
Comment (bitjammer) [inline lib/Syntax/ExprSyntax.cpp]: Can probably just use RC<TokenSyntax> here if you want.

**Review by bitjammer — APPROVED** [review]
Thanks HH!

**Pull request merged**